daemon/k8s_watcher.go

```diff
@@ -404,18 +404,22 @@ func (d *Daemon) serviceAddFn(obj interface{}) {
 		return
 	}
 
-	if strings.ToLower(svc.Spec.ClusterIP) == "none" || svc.Spec.ClusterIP == "" {
-		scopedLog.Info("Ignoring k8s service: headless")
+	if svc.Spec.ClusterIP == "" {
+		scopedLog.Info("Ignoring k8s service: empty ClusterIP")
 		return
 	}
 
 	svcns := types.K8sServiceNamespace{
-		Service:   svc.Name,
-		Namespace: svc.Namespace,
+		ServiceName: svc.Name,
+		Namespace:   svc.Namespace,
 	}
 
 	clusterIP := net.ParseIP(svc.Spec.ClusterIP)
-	newSI := types.NewK8sServiceInfo(clusterIP)
+	headless := false
+	if svc.Spec.ClusterIP == "None" {
```

Comment:
Do we need a `strings.ToLower()`?

Reply:
Only `None` is a valid value. `none` in clusterIP yields this error from kubectl: `spec.clusterIP: Invalid value: "none": must be empty, 'None', or a valid IP address`

Reply:
I understand. There is a common Robustness Principle or Postel's law which is typically followed in such cases:
Be conservative in what you do, be liberal in what you accept from others (often reworded as "Be conservative in what you send, be liberal in what you accept").

Reply:
Good point, done.